TYPO3.Flow/Classes/TYPO3/Flow/I18n/Service.php

```diff
@@ -279,6 +281,9 @@ protected function generateAvailableLocalesCollectionByScanningFilesystem()
             $directories = array(Files::getNormalizedPath($packageResourcesPath));
             while ($directories !== array()) {
                 $currentDirectory = array_pop($directories);
+                if ($blacklistPattern !== '' && preg_match('#' . str_replace('#', '\#', $blacklistPattern) . '#', $currentDirectory) === 1) {
```

Comment:
Maybe you can move the part building and escaping the pattern up to the other pattern building code (L271).
Would make this line easier to read, and it is also not needed per directory.